ml_metadata/tools/mlmd_bench/util.cc

```diff
@@ -58,8 +84,137 @@ tensorflow::Status GetExistingTypes(const FillTypesConfig& fill_types_config,
+      for (auto& context : get_response.contexts()) {
+        existing_nodes.push_back(context);
+      }
+      break;
+    }
+    default:
+      LOG(FATAL) << "Wrong specification for getting nodes in db!";
+  }
+  return tensorflow::Status::OK();
+}
+
+tensorflow::Status InsertTypesInDb(const int64 num_artifact_types,
+                                   const int64 num_execution_types,
+                                   const int64 num_context_types,
+                                   MetadataStore* store) {
+  PutTypesRequest put_request;
+  PutTypesResponse put_response;
+
+  for (int64 i = 0; i < num_artifact_types; i++) {
+    ArtifactType* curr_type = put_request.add_artifact_types();
+    curr_type->set_name(absl::StrCat("pre_insert_artifact_type-", i));
+    (*curr_type->mutable_properties())["property"] = STRING;
+  }
+
+  for (int64 i = 0; i < num_execution_types; i++) {
+    ExecutionType* curr_type = put_request.add_execution_types();
+    curr_type->set_name(absl::StrCat("pre_insert_execution_type-", i));
+    (*curr_type->mutable_properties())["property"] = STRING;
   }
+
+  for (int64 i = 0; i < num_context_types; i++) {
+    ContextType* curr_type = put_request.add_context_types();
+    curr_type->set_name(absl::StrCat("pre_insert_context_type-", i));
+    (*curr_type->mutable_properties())["property"] = STRING;
+  }
+
+  return store->PutTypes(put_request, &put_response);
+}
+
+tensorflow::Status InsertNodesInDb(const int64 num_artifact_nodes,
+                                   const int64 num_execution_nodes,
+                                   const int64 num_context_nodes,
+                                   MetadataStore* store) {
+  std::vector<Type> existing_artifact_types;
+  GetExistingTypes(/*specification=*/0, store, existing_artifact_types);
+  std::vector<Type> existing_execution_types;
+  GetExistingTypes(/*specification=*/1, store, existing_execution_types);
+  std::vector<Type> existing_context_types;
+  GetExistingTypes(/*specification=*/2, store, existing_context_types);
+
+  std::uniform_int_distribution<int64> uniform_dist_artifact_type_index{
+      0, (int64)(existing_artifact_types.size() - 1)};
+  std::uniform_int_distribution<int64> uniform_dist_execution_type_index{
+      0, (int64)(existing_execution_types.size() - 1)};
+  std::uniform_int_distribution<int64> uniform_dist_context_type_index{
+      0, (int64)(existing_context_types.size() - 1)};
+
+  std::minstd_rand0 gen(absl::ToUnixMillis(absl::Now()));
+
+  for (int64 i = 0; i < num_artifact_nodes; i++) {
+    const int64 type_index = uniform_dist_artifact_type_index(gen);
+    const string& node_name = absl::StrCat("pre_insert_artifact-", i);
+    PutArtifactsRequest put_request;
+    Artifact* curr_node = put_request.add_artifacts();
+    PrepareNode<ArtifactType, Artifact>(
+        node_name, absl::get<ArtifactType>(existing_artifact_types[type_index]),
+        *curr_node);
+    SetArtifactAdditionalFields(node_name, *curr_node);
```

Comment:
Artifact a0;
a0.set_uri("xxx")
for (int64 i = 0; i< ... ) {
    Artifact ai = a0;
    a0.set_name(i) ..
    a0.set_type (i..)
    a0.set_uri ...(i) ..
    PrepareNode(a0) ..
}